l10n_be_cooperator_national_number/models/subscription_request.py

```diff
@@ -25,30 +36,26 @@ def get_national_number_from_partner(self, partner):
 
     def validate_subscription_request(self):
         self.ensure_one()
-        if (
-            self.display_national_number
-            and not self.national_number
-            and not self.is_company
-        ):
+        if self.require_national_number and not self.national_number:
             raise UserError(_("National Number is required."))
         super().validate_subscription_request()
 
     def create_national_number(self, partner):
-        if self.display_national_number:
-            if not self.is_company:
-                values = {
-                    "name": self.national_number,
-                    "category_id": self.env.ref(
-                        "l10n_be_national_number.l10n_be_national_number_category"  # noqa
-                    ).id,
-                    "partner_id": partner.id,
-                }
-                self.env["res.partner.id_number"].create(values)
+        if not self.is_company:
+            values = {
+                "name": self.national_number,
+                "category_id": self.env.ref(
+                    "l10n_be_national_number.l10n_be_national_number_category"  # noqa
+                ).id,
+                "partner_id": partner.id,
+            }
+            self.env["res.partner.id_number"].create(values)
         return partner
 
     def create_coop_partner(self):
         partner = super().create_coop_partner()
-        self.create_national_number(partner)
+        if self.require_national_number:
```

Comment:
why putting a condition on "require" and not "display" here ? If the require config is unchecked but display is checked it will not write the national number on the partner ?

Reply:
Good call. I will patch this up, because simply switching to `display` won't suffice.

Reply:
This was surprisingly difficult to fix. Done now. Ready for another review.